internal/sanitize/node.go

```diff
@@ -66,6 +73,10 @@ func (n *Node) Sanitize(ctx context.Context) error {
 		}
 	}
 
+	if numMasters == 1 {
```

Comment:
Thanks for the updates Adrian! This check should happen before the if n.NoConcerns. Also we would need some tests to validate this feature. Tx!!

Reply:
Why should this happen before?

The `if n.NoConcerns` happens in the loop over the nodes, which is also required to count the master nodes. The handling of `numMasters == 1` can only happen after that loop has finished.